### PR TITLE
refactor(versioning/hashicorp): Enable strict null checks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Enable strict null checks

## Context:

- Ref: #7154

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]

**Comment by viceice** [comment]
Coverage issues

File: tsconfig.strict.json [hunk 1]
```diff
@@ -387,11 +387,7 @@
     "lib/util/package-rules.ts",
     "lib/util/template/index.ts",
     "lib/versioning/api.ts",
-    "lib/versioning/aws-machine-image/index.ts",
```
Comment (rarkins) [inline tsconfig.strict.json:390]: Bad merge?

Comment (zharinov) [reply tsconfig.strict.json:390]: No, just marking other files ready in bulk

Comment (rarkins) [reply tsconfig.strict.json:390]: Were they missed in their own PRs? I.e. already compliant?

**Review by rarkins — APPROVED** [review]

**Pull request merged**